src/common/transformations/src/transformations/mlir/mlir_op.cpp

```diff
@@ -225,12 +241,25 @@ std::unique_ptr<llvm::Module> lowerToLLVMIR(Operation* module, llvm::LLVMContext
 struct MemRefDescriptor {
     MemRefDescriptor() = default;
 
-    MemRefDescriptor    (ov::Tensor tensor)
+    MemRefDescriptor    (ov::Tensor tensor, ov::Shape module_input_shape)
         : allocated(tensor.data()),
           aligned(tensor.data()),
           offset(0),
-          shape(tensor.get_shape().begin(), tensor.get_shape().end()) {
-        strides.resize(tensor.get_shape().size());
+          shape(module_input_shape.begin(), module_input_shape.end()) {
+        if (shape.size() != tensor.get_shape().size()) {
+            // validate that the shape difference is due to trailing '1's
+            for (size_t i = 0; i < shape.size(); ++i) {
+                if (shape[i] != tensor.get_shape()[i]) {
+                    OPENVINO_THROW("Mismatch in shape sizes");
+                }
+            }
+            for (size_t i = shape.size(); i < tensor.get_shape().size(); ++i) {
+                if (tensor.get_shape()[i] != 1) {
+                    OPENVINO_THROW("Mismatch in shape sizes");
+                }
+            }
+        }
+        strides.resize(shape.size());
```

Comment:
This is needed due to the fact that GPU memory formats could hold at least 4-dimensions, causing trailing extra dims (`<64x128x1x1>` instead of `<64x128>`). This code compares input tensors' dimensions with the input dimensions of a MLIR module and trims extra dims.